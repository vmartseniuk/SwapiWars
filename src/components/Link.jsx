/* eslint-disable react/prop-types */
import React from "react";
import { Link as MaterialLink } from "@material-ui/core";
import { Link as RouterLink } from "react-router-dom";

const Link = ({ children, ...props }) => (
  <MaterialLink component={RouterLink} {...props}>
    {children}
  </MaterialLink>
);

const ExternalLink = ({ children, ...props }) => {
  return (
    <MaterialLink {...props} component="a" target="_blank">
      {children}
    </MaterialLink>
  );
};

export { Link, ExternalLink };
