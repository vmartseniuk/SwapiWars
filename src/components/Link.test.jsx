import { create } from "react-test-renderer";

import { ExternalLink } from "./Link";

describe("Links", () => {
  it("ExternalLink renders correctly", () => {
    const rendered = create(<ExternalLink to="/">test link</ExternalLink>);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
