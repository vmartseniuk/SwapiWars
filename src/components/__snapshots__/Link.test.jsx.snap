// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links ExternalLink renders correctly 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  onBlur={[Function]}
  onFocus={[Function]}
  target="_blank"
  to="/"
>
  test link
</a>
`;
